### DISCUSSION TITLE
Development Primer

### DISCUSSION THREAD

**Discussion by digitalnoiz** [ebullient/ttrpg-convert-cli#330] [Q&A] [answered]
Greetings,

I came across this project while trying to build a DM obsidian vault. It Fantastic! And there's bugs! and that's fine... I'm intrigued to try addressing some of the problems that bother me but I'm a lowly C# developer with a pittance of  Java experience. However if I had a debugger I think I could make a go at it!

So I'm wondering if anyone might have:
1.  A good launch.json configuration for this project that can set up debugging in visual studio code
2. Information on how I can define arguments for a config path, output path and 5etools input path so I can run with a defined input
3. Any advice to prevent the javac compiler on windows from changing line endings on all the source files of the project when I build

Any assistance would be greatly appreciated! -D

**Comment by ebullient** [comment, marked as answer]
I added some content to [CONTRIBUTING.md](https://github.com/ebullient/ttrpg-convert-cli/blob/main/CONTRIBUTING.md), we'll see if that helps.

I did add a rough flow, and some tips for how to run tests for subsets of data. Indexing always happens for everything... ingest is one big mammoth step, as you need all of the bits there for rendering.

My advice: get VSCode set up (the guide recommends plugins), and add Copilot. Ask Copilot to explain anything you don't understand (due to syntax or convention differences)

**Reply by digitalnoiz** [reply]
Thanks for the additional info.  I still haven't fully tamed the tools to do what I want but the test tools should help. For line endings I turned on autocrlf in git. I usually keep it off for hybrid .net/node/javascript web projects as it always seems to cause problems with node files. For this project, when ever I ran compile tasks, the line endings of the .java files gets changed. I'm assuming it's the JVM trying to accommodate the host OS or something and I was assuming I could find a flag to change the behavior but haven't come across a direct solution yet so managing it in git it is.

**Reply by ebullient** [reply]
editorconfig plugin will make your editor stop doing the wrong line ending

**Reply by ebullient** [reply]
and there is a plugin that will reformat things (and remove stray unused imports and other things), but I've never seen that plugin change line endings before (unless git checked them out with crlf?)
